cypress/e2e/e2e/engageE2e.cy.js

```diff
@@ -0,0 +1,228 @@
+
+  it('Maintains selected filter after returning from customer details', () => {
+    cy.get('@acceptButton').click();
+
+    // Verify the size filter exists and is visible
+    cy.get('@sizeFilter').should('be.visible');
+
+    // Select "Small" from the size filter
+    cy.get('@sizeFilter').select('Small');
+
+    // Verify that "Small" is now selected
+    cy.get('@sizeFilter').within(() => {
+      cy.get('option:selected').should('have.value', 'Small');
+    });
+
+    // Navigate to the customer details page (clicking on the last cell of the first row)
+    cy.get('tbody > :nth-child(1) > :nth-child(6)').click();
+
+    // Go back to the customer list page
+    cy.get('.Button_buttonContainer__ti4sL > button').click();
+
+    // Verify that "Small" is still selected in the size filter
+    cy.get('@sizeFilter').within(() => {
+      cy.get('option:selected').should('have.value', 'Small');
+    });
+  });
+
+  it('Verify footer links', () => {
+    // Click the accept button before verifying the footer
+    cy.get('@acceptButton').click();
+
+    // Verify the footer is present and visible on the page
+    cy.get('[data-testid="footer"]').should('be.visible');
+
+    // Verify the footer contains the correct copyright text
+    cy.get('[data-testid="footer"] > p').should(
+      'have.text',
+      'Copyright 2025 - Talking About Testing'
+    );
+
+    // Define an array of expected footer links (text and href values)
+    const links = [
+      { text: 'Blog', href: 'https://talkingabouttesting.com' },
+      { text: 'Courses', href: 'https://talking-about-testing.vercel.app/' },
+      {
+        text: 'Podcast',
+        href: 'https://open.spotify.com/show/5HFlqWkk6qtgJquUixyuKo',
+      },
+      { text: 'YouTube', href: 'https://youtube.com/@talkingabouttesting' },
+    ];
+
+    // Loop through each link in the array and verify its properties
+    links.forEach(({ text, href }) => {
+      // Find the link inside the footer by its text
+      cy.contains('[data-testid="footer"] a', text)
+        // Verify the link has the correct 'href' attribute
+        .should('have.attr', 'href', href)
+        // Ensure the link opens in a new tab by checking 'target="_blank"'
+        .and('have.attr', 'target', '_blank');
+    });
+  });
+
+  it('Empty Hi there', () => {
+    // Click the accept button before running assertions
+    cy.get('@acceptButton').click();
+
+    // Verify that the element is empty
+    cy.get('[data-testid="name"]').should('be.empty');
+
+    // Verify that the element contains "Hi there"
+    cy.get('[data-testid="greeting-header"]').should('contain', 'Hi there');
+  });
+
+  it('Hi Joe there', () => {
+    // Click the accept button before running assertions
+    cy.get('@acceptButton').click();
+
+    // Write Joe on the name field
+    cy.get('[data-testid="name"]').type('Joe');
+
+    // Verify that the element contains "Hi there"
+    cy.get('[data-testid="greeting-header"]').should('contain', 'Hi Joe');
```

Comment:
A mesma coisa se aplica aqui.

Poderia ser:

```suggestion
    cy.contains('h2', 'Hi Joe').should('be.visible')
```